sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MethodsShouldUseBaseTypes.cs

```diff
@@ -153,11 +151,9 @@ private ITypeSymbol FindParameterUseAsType(IdentifierNameSyntax identifier, Sema
                     return HandlePropertyOrField(identifier, accessedMember);
                 }
 
-                var invocationExpression = conditionalAccess.WhenNotNull as InvocationExpressionSyntax;
-                if (invocationExpression != null)
+                if (conditionalAccess.WhenNotNull is InvocationExpressionSyntax invocationExpression)
                 {
-                    var memberBinding = invocationExpression.Expression as MemberBindingExpressionSyntax;
-                    if (memberBinding != null)
+                    if (invocationExpression.Expression is MemberBindingExpressionSyntax memberBinding)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1066)